plugins/common/socket/stream.go

```diff
@@ -330,12 +352,18 @@ func (l *streamListener) read(conn net.Conn, onData CallbackData) error {
 			break
 		}
 
+		receiveTime := time.Now()
 		src := conn.RemoteAddr()
 		if l.path != "" {
 			src = &net.UnixAddr{Name: l.path, Net: "unix"}
 		}
+
 		data := scanner.Bytes()
-		onData(src, data)
+		d := make([]byte, len(data))
+		copy(d, data)
```

Comment:
Why copying again?

Reply:
In this case I was not 100% sure it is needed, because I don't entirely understand how the scanner works.

Reply:
Actually, it is also needed here. According to the docs: https://pkg.go.dev/bufio#Scanner.Bytes

> The underlying array may point to data that will be overwritten by a subsequent call to Scan. It does no allocation.